### PR TITLE
refine accountability contract

### PR DESCRIPTION
**Pull request by lorenzo-dev1**
Closes https://github.com/autonity/autonity/issues/823, partially address https://github.com/autonity/autonity/issues/1050

### CODE REVIEW

**Review by ScottRMalley — COMMENTED** [review]
Overall, definitely feels like a good improvement. Just one small concern.

File: autonity/solidity/contracts/Accountability.sol [hunk 1]
```diff
@@ -423,49 +414,106 @@ contract Accountability is IAccountability {
...
+        Setters
+    ============================================================
+    */
+
+    /**
+    * @notice setEpochPeriod, called by the AC when there is an epoch period change
+    * @dev restricted to the autonity contract
+    * @param _newPeriod, the new epoch period
+    */
+    function setEpochPeriod(uint256 _newPeriod) external virtual onlyAutonity {
```
[the remark this answers is not in the document]
Comment (lorenzo-dev1) [reply autonity/solidity/contracts/Accountability.sol]: good point, we don't need this mechanism. Fixed in https://github.com/autonity/autonity/pull/1064/commits/4a9fd58add58541aff0d7d426cbf766fc1071211

**Review by vpiyush — APPROVED** [review]
LGTM!!

**Review by ScottRMalley — APPROVED** [review]
Looks good

**Review by Jason-Zhangxin-Chen — APPROVED** [review]
Looks good!

**Pull request merged**